safe_control_gym/controllers/lqr/ilqr.py

```diff
@@ -57,13 +59,14 @@ def __init__(
         self.env = env_func(info_in_reset=True, done_on_out_of_bound=True)
 
         # Controller params.
-        self.model = self.env.symbolic
+        self.prior_info = prior_info
+        self.model = self.get_prior(self.env)
```

Comment:
in a convo w justin I think we decided that self.get_prior should be called in reset() (as well as init if thats easier). Essentially this is b/c if there is prior randomization it should re-randomize every time the controller or safety filter is reset.

Reply:
i think here `self.model` is used to get other information to initialize the controller (e.g. `nx`, `nu`, etc). But Federico is right, feel free to call `get_prior()` in either `__init__()` or `reset()` that you find fit for the specific controller.